File: AtlasWeb/Core/Modules/Tibbi_Surec/Yogun_Bakim_Modulu/BallardPhysicalScoreFormViewModel.ts
//$FB278BF0
import { BaseViewModel } from "NebulaClient/Model/BaseViewModel";
import { BallardPhysical } from "NebulaClient/Model/AtlasClientModel";
import { Type } from "../../../wwwroot/app/NebulaClient/ClassTransformer";

export class BallardPhysicalScoreFormViewModel extends BaseViewModel {
    @Type(() => BallardPhysical)
    public _BallardPhysical: BallardPhysical = new BallardPhysical();
}

File: AtlasWeb/Core/Modules/Tibbi_Surec/Yogun_Bakim_Modulu/BallardPhysicalScoreFormViewModel.cs
//$D06884E5
using System;
using System.Linq;
using Core.Models;
using Infrastructure.Filters;
using Infrastructure.Models;
using TTInstanceManagement;
using TTObjectClasses;
using TTDefinitionManagement;
using TTUtils;

namespace Core.Controllers
{
    public partial class BallardPhysicalServiceController
    {
        partial void PreScript_BallardPhysicalScoreForm(BallardPhysicalScoreFormViewModel viewModel, BallardPhysical ballardPhysical, TTObjectContext objectContext)
        {
            EpisodeAction episodeAction = ballardPhysical.EpisodeAction;
            if (episodeAction == null)
            {
                Guid? selectedEpisodeActionObjectID = viewModel.GetSelectedEpisodeActionID();
                if (selectedEpisodeActionObjectID.HasValue)
                {
                    episodeAction = objectContext.GetObject<EpisodeAction>(selectedEpisodeActionObjectID.Value);
                    ballardPhysical.EpisodeAction = episodeAction;
                }
            }
            ballardPhysical.EntryUser = Common.CurrentResource;
        }
    }
}

namespace Core.Models
{
    public partial class BallardPhysicalScoreFormViewModel: BaseViewModel
    {
    }
}
